src/path/index.ts

```diff
@@ -116,52 +163,68 @@ export function toPosix(
+export function combine<P extends Partition>(a: DirectoryPath<Partitioned<P>>, b: FilePath<SegmentsNonEmpty>): FilePath<PartitionedNonEmpty<P>>
+export function combine<P extends Partition>(a: DirectoryPath<Partitioned<P>>, b: FilePath<Segments>): FilePath<Partitioned<P>>
+export function combine(a: DirectoryPath<Segments>, b: FilePath<Segments>): FilePath<Segments>
+export function combine<P extends Partition>(a: DirectoryPath<PartitionedNonEmpty<P>>, b: DirectoryPath<Segments>): DirectoryPath<PartitionedNonEmpty<P>>
+export function combine<P extends Partition>(a: DirectoryPath<Partitioned<P>>, b: DirectoryPath<SegmentsNonEmpty>): DirectoryPath<PartitionedNonEmpty<P>>
+export function combine<P extends Partition>(a: DirectoryPath<Partitioned<P>>, b: DirectoryPath<Segments>): DirectoryPath<Partitioned<P>>
+export function combine(a: DirectoryPath<Segments>, b: DirectoryPath<Segments>): DirectoryPath<Segments>
+export function combine<P extends Partition>(a: DirectoryPath<PartitionedNonEmpty<P>>, b: DistinctivePath<Segments>): DistinctivePath<PartitionedNonEmpty<P>>
+export function combine<P extends Partition>(a: DirectoryPath<Partitioned<P>>, b: DistinctivePath<SegmentsNonEmpty>): DistinctivePath<PartitionedNonEmpty<P>>
+export function combine<P extends Partition>(a: DirectoryPath<Partitioned<P>>, b: DistinctivePath<Segments>): DistinctivePath<Partitioned<P>>
+export function combine(a: DirectoryPath<Segments>, b: DistinctivePath<Segments>): DistinctivePath<Segments>
+export function combine(a: DirectoryPath<Segments>, b: DistinctivePath<Segments>): DistinctivePath<Segments> {
   return map(p => unwrap(a).concat(p), b)
 }
 
 /**
- * Is this `DistinctivePath` of the given `Branch`?
+ * Is this `DistinctivePath` a directory?
  */
-export function isBranch(branch: Branch, path: DistinctivePath): boolean {
-  return unwrap(path)[ 0 ] === branch
+export function isDirectory<P>(path: DistinctivePath<P>): path is DirectoryPath<P> {
+  return !!(path as DirectoryPath<P>).directory
 }
 
 /**
- * Is this `DistinctivePath` a directory?
+ * Is this `DistinctivePath` a file?
  */
-export function isDirectory(path: DistinctivePath): path is DirectoryPath {
-  return !!(path as DirectoryPath).directory
+export function isFile<P>(path: DistinctivePath<P>): path is FilePath<P> {
+  return !!(path as FilePath<P>).file
 }
 
 /**
- * Is this `DistinctivePath` a file?
+ * Is this `DistinctivePath` of the given `Partition`?
+ */
+export function isPartition(partition: Partition, path: DistinctivePath<Segments>): boolean {
+  return unwrap(path)[ 0 ] === partition
+}
+
+/**
+ * Is this `DistinctivePath` of the given `RootBranch`?
  */
-export function isFile(path: DistinctivePath): path is FilePath {
-  return !!(path as FilePath).file
+export function isRootBranch(rootBranch: RootBranch, path: DistinctivePath<Segments>): boolean {
```

Comment:
The name `isRootBranch` makes me think we are checking if a path is the `private` or `public` directory. What would you think about renaming this to something like `isOnRootBranch`?

Reply:
Sounds good, will rename!